### ISSUE TITLE
ArrayIndexOutOfBoundsException thrown from HttpConnection.get

### ISSUE THREAD

**Issue by edashman** [jhy/jsoup#1092]
We recently encountered a site `https://www.vivre-a-niort.com` with headers containing `text/html; charset=utf-8â ` which results in an `ArrayIndexOutOfBoundsException` thrown from `HttpConnection.looksLikeUtf8(byte[] input)`. The offending character seems to be `â` which leads the method to incorrectly determine the end of the byte array.

The exception thrown is
```
java.lang.ArrayIndexOutOfBoundsException: 26

	at org.jsoup.helper.HttpConnection$Base.looksLikeUtf8(HttpConnection.java:437)
	at org.jsoup.helper.HttpConnection$Base.fixHeaderEncoding(HttpConnection.java:400)
	at org.jsoup.helper.HttpConnection$Base.addHeader(HttpConnection.java:386)
	at org.jsoup.helper.HttpConnection$Response.processResponseHeaders(HttpConnection.java:1075)
	at org.jsoup.helper.HttpConnection$Response.setupFromConnection(HttpConnection.java:1019)
	at org.jsoup.helper.HttpConnection$Response.execute(HttpConnection.java:752)
	at org.jsoup.helper.HttpConnection$Response.execute(HttpConnection.java:773)
	at org.jsoup.helper.HttpConnection$Response.execute(HttpConnection.java:773)
	at org.jsoup.helper.HttpConnection$Response.execute(HttpConnection.java:722)
	at org.jsoup.helper.HttpConnection.execute(HttpConnection.java:306)
	at org.jsoup.helper.HttpConnection.get(HttpConnection.java:295)
```
and can be reproduced using the following test
```
    @Test
    public void test() throws IOException {
        Connection connect = Jsoup.connect("https://www.vivre-a-niort.com");
        connect.get();
    }
```
while using the latest version of jsoup `1.11.3`

**Comment by jhy** [comment]
I'm sorry, but I didn't to this earlier, and now the site doesn't return that. I can't replicate with by creating a string matching that. Is anyone still seeing this issue somewhere?

**Comment by edashman** [comment]
i can no longer reproduce this locally on version `1.11.3` even though the site is still returning `content-type` as `text/html; charset=utf-8â`

**Comment by jhy** [comment]
I think this is actually a dupe of #1172 which I fixed in 1.12.1. Can you double check what version you're running?
